core/src/workflows/respond_validation_package_request.rs

```diff
@@ -0,0 +1,47 @@
+use crate::{
+    action::{Action, ActionWrapper, DirectMessageData},
+    context::Context,
+    instance::dispatch_action,
+    network::direct_message::DirectMessage,
+    nucleus::actions::build_validation_package::build_validation_package,
+};
+
+use holochain_core_types::{cas::content::Address, entry::Entry, error::HolochainError};
+use std::{convert::TryFrom, sync::Arc};
+
+fn get_entry(address: &Address, context: &Arc<Context>) -> Result<Entry, HolochainError> {
+    let raw = context
+        .state()
+        .unwrap()
+        .agent()
+        .chain()
+        .content_storage()
+        .read()
+        .unwrap()
+        .fetch(address)?
+        .ok_or(HolochainError::ErrorGeneric("Entry not found".to_string()))?;
+
+    Entry::try_from(raw)
+}
+
+pub async fn respond_validation_package_request(
+    to_agent_id: Address,
+    msg_id: String,
+    requested_entry_address: Address,
+    context: Arc<Context>,
+) {
+    let maybe_validation_package = match get_entry(&requested_entry_address, &context) {
+        Ok(entry) => await!(build_validation_package(&entry, &context)).ok(),
+        Err(_) => None,
```

Comment:
Do we really send a None value back across the network if there was an error?  I have the feeling that we at least have to tell the requester that we had some kind of error...  followup?